tests/test_ledger_attachments.py

```diff
@@ -76,4 +102,36 @@ def sort_dict_values(items):
     engine._client.post("journal/update_attachments.json", data={'id': ledger_ids[1], 'fileIds': file_ids})
     engine._client.invalidate_journal_cache()
     expected = initial | {ledger_ids[0]: ['/file1.txt'], ledger_ids[1]:  ['/file1.txt', '/subdir/file2.txt']}
-    assert sort_dict_values(engine._get_ledger_attachments()) == sort_dict_values(expected)
\ No newline at end of file
+    assert sort_dict_values(engine._get_ledger_attachments()) == sort_dict_values(expected)
+
+def test_attach_ledger_files(files, ledger_attached_ids):
+    engine = CashCtrlLedger()
+    # Attach file that should be deleted
+    engine._client.post("journal/update_attachments.json", data={'id': ledger_attached_ids[0], 'fileIds': files['id'].iat[0]})
+    # Attach file that should be updated
+    engine._client.post("journal/update_attachments.json", data={'id': ledger_attached_ids[1], 'fileIds': files['id'].iat[0]})
+    # Attach file that should left untouched
+    engine._client.post("journal/update_attachments.json", data={'id': ledger_attached_ids[2], 'fileIds': files['id'].iat[0]})
+
+    # Update attachments with detach=False
+    engine.attach_ledger_files(detach=False)
+    attachments = engine._get_ledger_attachments()
+    attachments = {k: v for k, v in attachments.items() if k in ledger_attached_ids}
+    expected = {
+        ledger_attached_ids[0]: ['/file1.txt'],
+        ledger_attached_ids[1]: ['/subdir/file2.txt'],
+        ledger_attached_ids[2]: ['/file1.txt'],
+        ledger_attached_ids[3]: ['/file1.txt'],
+    }
+    assert sort_dict_values(expected) == sort_dict_values(attachments)
+
+    # Update attachments with detach=True
+    engine.attach_ledger_files(detach=True)
+    attachments = engine._get_ledger_attachments()
+    attachments = {k: v for k, v in attachments.items() if k in ledger_attached_ids}
+    expected = {
+        ledger_attached_ids[1]: ['/subdir/file2.txt'],
+        ledger_attached_ids[2]: ['/file1.txt'],
+        ledger_attached_ids[3]: ['/file1.txt'],
+    }
+    assert sort_dict_values(expected) == sort_dict_values(attachments)
```

Comment:
Please add further test cases:
- include reference paths that do not match remote files.
- include a ledger entry with multiple attachments